scripts/nb-tester/qiskit_docs_notebook_tester/__init__.py

```diff
@@ -67,15 +79,15 @@ def filter_paths(paths: list[Path], args: argparse.Namespace, config: Config) ->
             )
             continue
 
-        if not submit_jobs and matches(path, config.notebooks_that_submit_jobs):
+        if not args.submit_jobs and matches(path, config.notebooks_no_mock):
             print(
-                f"ℹ️ Skipping {path} as it submits jobs; use the --submit-jobs flag to run it."
+                f"ℹ️ Skipping {path} as it doesn't work with mock hardware; use the --submit-jobs flag to run it."
             )
             continue
 
-        if args.only_submit_jobs and not matches(path, config.notebooks_that_submit_jobs):
+        if args.only_submit_jobs and not matches(path, config.notebooks_that_submit_jobs+config.notebooks_no_mock):
```

Comment:
Nit: better to not concatenate the two collections repeatedly inside the for loop. Instead, define a variable outside of the for loop